sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java

```diff
@@ -236,8 +236,15 @@ public UnboundedJmsSource(
     public List<UnboundedJmsSource> generateInitialSplits(
         int desiredNumSplits, PipelineOptions options) throws Exception {
       List<UnboundedJmsSource> sources = new ArrayList<>();
-      for (int i = 0; i < desiredNumSplits; i++) {
+      if (topic != null) {
```

Comment:
Is it reasonable to add a unit test that fails before this change and passes after this change, just to make sure we don't regress in the future?